webrender/src/platform/unix/font.rs

```diff
@@ -153,93 +166,81 @@ impl FontContext {
         };
 
         let result = unsafe { FT_Render_Glyph(slot, render_mode) };
-        if !result.succeeded() {
-            error!("Unable to rasterize {:?} with {:?}", key, render_mode);
+        if result != SUCCESS {
+            error!("Unable to rasterize {:?} with {:?}, {:?}", key, render_mode, result);
             return None;
         }
 
+        let dimensions = Self::get_glyph_dimensions_impl(slot).unwrap();
         let bitmap = unsafe { &(*slot).bitmap };
-        let metrics = unsafe { &(*slot).metrics };
-        let glyph_width = (metrics.width >> 6) as i32;
-        let glyph_height = (metrics.height >> 6) as i32;
-        let mut final_buffer = Vec::with_capacity(glyph_width as usize *
-                                                  glyph_height as usize *
+        let pixel_mode = unsafe { mem::transmute(bitmap.pixel_mode as u32) };
+        info!("Rasterizing {:?} as {:?} with dimensions {:?}", key, render_mode, dimensions);
+        // we may be filling only a part of the buffer, so initialize the whole thing with 0
```

Comment:
this is a leftover comment, I'll remove it when applying the other fixes a reviewer may request